test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs

```diff
@@ -102,9 +102,11 @@ public void ObserverCallbackExceptionTest()
         }
 
         [Theory]
-        [InlineData(AggregationTemporality.Cumulative)]
-        [InlineData(AggregationTemporality.Delta)]
-        public void StreamNamesDuplicatesAreNotAllowedTest(AggregationTemporality temporality)
+        [InlineData(AggregationTemporality.Cumulative, true)]
+        [InlineData(AggregationTemporality.Cumulative, false)]
+        [InlineData(AggregationTemporality.Delta, true)]
+        [InlineData(AggregationTemporality.Delta, false)]
+        public void StreamNamesDuplicatesAreNotAllowedTest(AggregationTemporality temporality, bool hasView)
```

Comment:
Rename this to `StreamNameDuplicatesAreAllowedWithDifferentMetersTest`?

We should add another unit test to confirm that we cannot add duplicate metric stream names when the instruments created from the same meter.

Reply:
This unit test currently tests both

1) that duplicate names are not allowed in when the instruments are created from the same meter, and
2) that duplicate names are allowed when instruments are from different meters

Are you suggesting breaking this into two tests for clarity? If so, I'm 👍 on that idea.

Reply:
> We should also add another unit test to confirm that we cannot add duplicate metric stream names when the instruments created from the same meter.

Ohh I just realized that this was already covered before this PR. I guess we could have two different unit tests so that their names match what they are testing.

Reply:
Split em up. ✂️